src/main/java/org/spongepowered/api/GameRegistry.java

```diff
@@ -536,4 +543,110 @@
      */
     List<BannerPatternShape> getBannerPatternShapes();
 
+    /**
+     * Gets the {@link TextColor} with the provided name.
+     *
+     * @param name The name of the text color
+     * @return The {@link TextColor} with the given name or Optional.absent() if not found
+     */
+    Optional<TextColor> getTextColor(String name);
+
+    /**
+     * Gets a {@link List} of all possible {@link TextColor}s.
+     *
+     * @return The list of all available {@link TextColor}s
+     */
+    List<TextColor> getTextColors();
+
+    /**
+     * Gets the {@link TextStyle} with the provided name.
+     *
+     * @param name The name of the text style
+     * @return The {@link TextStyle} with the given name or Optional.absent() if not found
+     */
+    Optional<TextStyle> getTextStyle(String name);
+
+    /**
+     * Gets a {@link List} of all possible {@link TextStyle}s.
+     *
+     * @return The list of all available {@link TextStyle}s
+     */
+    List<TextStyle> getTextStyles();
+
+    /**
+     * Gets the {@link ChatType} with the provided name.
+     *
+     * @param name The name of the chat type
+     * @return The {@link ChatType} with the given name or Optional.absent() if not found
+     */
+    Optional<ChatType> getChatType(String name);
+
+    /**
+     * Gets a {@link List} of all possible {@link TextStyle}s.
+     *
+     * @return The list of all available {@link TextStyle}s
+     */
+    List<ChatType> getChatTypes();
+
+    /**
+     * Gets the {@link SelectorType} with the provided name.
+     *
+     * @param name The name of the selector type
+     * @return The {@link SelectorType} with the given name or Optional.absent() if not found
+     */
+    Optional<SelectorType> getSelectorType(String name);
```

Comment:
Yea, that's what you get for merging my commits with @Minecrell. Just a small merge error.